packages/extension/src/background/transaction/actions.ts

```diff
@@ -160,17 +160,12 @@ export function getLedgerFromMixedGenesis(genesisId: string, genesisHash: string
       }
     }
     
-    // Injected networks may have additional information, multiples, or additional checks
-    // so we will check them separately
+    // Injected networks may have additional validations so we check them separately
     const injectedNetworks = Settings.getCleansedInjectedNetworks();
-    injectedNetworks.forEach(network => {
-      if (network['genesisId'] === genesisId) {
-        // Found genesisId, make sure the hash matches 
-        if (!genesisHash || genesisHash === network.genesisHash) { 
-          return network;
-        }
-      }
-    });
+    ledger = injectedNetworks.find((network) => network['genesisId'] === genesisId);
```

Comment:
The ARCs requires the ID and the hash both to match if the match is provided.  So should not return if the hash is different and should continue the checks to allow for a duplicate genesisId with a different hash.

Reply:
same as below's comment